tests/test_x509_ext.py

```diff
@@ -1147,6 +1147,43 @@ def test_ne(self):
         assert gn != object()
 
 
+class TestOtherName(object):
+    def test_invalid_args(self):
+        with pytest.raises(TypeError):
+            x509.OtherName(b"notanobjectidentifier", b"derdata")
+
+        with pytest.raises(TypeError):
+            x509.OtherName(x509.ObjectIdentifier("1.2.3.4"), "notderdata")
+
+    def test_repr(self):
+        gn = x509.OtherName(x509.ObjectIdentifier("1.2.3.4"), b"derdata")
+        assert repr(gn) == (
+            "<OtherName(type_id=<ObjectIdentifier(oid=1.2.3.4, "
+            "name=Unknown OID)>, value=b'derdata')>"
```

Comment:
This will need a branch on `six.PY3` since there won't be a `b` prefix on python2 (there's a few other examples of this around)